services/packagist/packagist-base.js

```diff
@@ -73,16 +94,74 @@ class BasePackagistService extends BaseJsonService {
     })
   }
 
-  getDefaultBranch(json, user, repo) {
-    const packageName = this.getPackageName(user, repo)
-    return Object.values(json.packages[packageName]).find(
-      b => b['default-branch'] === true
-    )
-  }
-
   getPackageName(user, repo) {
     return `${user.toLowerCase()}/${repo.toLowerCase()}`
   }
+
+  /**
+   * Extract the array of minified versions of the given packageName,
+   * expand them back to their original format then return.
+   *
+   * @param {object} json The response of Packagist v2 API.
+   * @param {string} packageName The package name.
+   *
+   * @returns {object[]} An array of version metadata object.
+   *
+   * @see https://github.com/composer/metadata-minifier/blob/c549d23829536f0d0e984aaabbf02af91f443207/src/MetadataMinifier.php#L16-L46
+   */
+  static expandPackageVersions(json, packageName) {
+    const versions = json.packages[packageName]
+    const expanded = []
+    let expandedVersion = null
+
+    for (const i in versions) {
+      const versionData = versions[i]
+      if (!expandedVersion) {
+        expandedVersion = { ...versionData }
+        expanded.push(expandedVersion)
+        continue
+      }
+
+      expandedVersion = { ...expandedVersion, ...versionData }
+      for (const key in expandedVersion) {
+        if (expandedVersion[key] === '__unset') {
+          delete expandedVersion[key]
+        }
+      }
+      expanded.push(expandedVersion)
+    }
+
+    return expanded
+  }
+
+  /**
+   * Find the object representation of the latest release.
+   *
+   * @param {object[]} versions An array of object representing a version.
+   * @param {boolean} includePrereleases Includes pre-release semver for the search.
+   *
+   * @returns {object} The object of the latest version.
+   * @throws {NotFound} Thrown if there is no item from the version array.
+   */
+  findLatestRelease(versions, includePrereleases = false) {
```

Comment:
Was going to ask if it'd be worth adding some unit tests for this function since the logic, though not terribly complex, isn't exactly trivial either.

I see that we're indirectly testing it with tests that targetting some of the child classes. That works for me, at least as long as we don't end up duplicating tests down the road

Reply:
Add more testing on the latest commit. Please check.